src/clarity-angular/datagrid/datagrid-column-toggle.spec.ts

```diff
@@ -0,0 +1,306 @@
+/*
+ * Copyright (c) 2016-2017 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import { Component, QueryList, TemplateRef, ViewChildren } from "@angular/core";
+import { TestContext } from "./helpers.spec";
+import { HideableColumnService } from "./providers/hideable-column.service";
+import { DatagridColumnToggle } from "./datagrid-column-toggle";
+import { DatagridHideableColumn } from "./datagrid-hideable-column";
+
+export default function (): void {
+    describe("Datagrid Column Toggle component", function () {
+        describe("Typescript API", function () {
```

Comment:
Do we test whether the same column keeps its previous state such as style, classes etc when it gets hidden and redisplayed again? Because in the app demo, when I use toggle buttons to hide and open the same column, their strict width classes are lost.